### PR TITLE
new future: init() initializing a param field with a non-param value

### PR DESCRIPTION
**Pull request by cassella**
This should fail to compile.

This is the future for issue #8406

### CODE REVIEW

File: test/classes/initializers/generics/init-param-from-non-param.chpl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+config var notaparam = 77;
```
Comment (lydia-duncan) [inline test/classes/initializers/generics/init-param-from-non-param.chpl]: Not a bad thing, but I'm curious why you went with this program as opposed to using the local var you mentioned in the original issue

Comment (cassella) [reply test/classes/initializers/generics/init-param-from-non-param.chpl]: In the original issue, I was hung up on the ``record`` type, and the local var was just there to separate the param initialization from the ``new R()``.  When I found the issue applied to ``int`` as well, I used a ``config var`` so I didn't have to contemplate whether the compiler believes the value below is "known at compile time" as the error message says:

```chapel
var notaparam = 7;
param p = notaparam;
```

With a config var, there's no sense in which it's known at compile time.

Comment (lydia-duncan) [reply test/classes/initializers/generics/init-param-from-non-param.chpl]: I can see that logic, but a local variable is a more basic case (assuming it still causes the same failure) and really should be treated reasonably similarly by the compiler.

Comment (cassella) [reply test/classes/initializers/generics/init-param-from-non-param.chpl]: Ok.  I'll try making that change tonight.

**Pull request merged**